pos_sale_order/models/sale_order.py

```diff
@@ -206,6 +206,11 @@ def action_pos_order_invoice(self):
     def _prepare_invoice(self):
         res = super()._prepare_invoice()
         res["session_id"] = self.session_id.id
+        if self.config_id.cash_rounding and (
+            not self.config_id.only_round_cash_method
+            or any(p.payment_method_id.is_cash_count for p in self.payment_ids)
```

Comment:
it's not the good algorithm, we should take the one from odoo/point_of_sale : round only if the last of the payment_method_id is cash_count